test/integration/targets/ec2_snapshot/tasks/main.yml

```diff
@@ -0,0 +1,250 @@
+---
+# Tests for EC2 Snapshot
+#
+# Tests ec2_snapshot:
+# - Snapshot creation
+# - Create with last_snapshot_min_age
+# - Snapshot deletion
+#
+# Tests ec2_snapshot_info:
+# - Listing snapshots for filter: tag
+#
+# Possible Bugs:
+# - check_mode not supported
+#
+- name: Integration testing for ec2_snapshot
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+
+  block:
+    - ec2_ami_info:
+        owners: amazon
+        filters:
+          architecture: x86_64
+          virtualization-type: hvm
+          root-device-type: ebs
+          name: "amzn-ami-hvm*"
+      register: amis
+
+    - name: Setup an instance for testing
+      ec2_instance:
+        name: '{{ resource_prefix }}'
+        instance_type: t2.nano
+        image_id: "{{ (amis.images | sort(attribute='creation_date') | last).image_id }}"
+        wait: yes
+        volumes:
+          - device_name: /dev/xvda
+            ebs:
+              volume_size: 8
+              delete_on_termination: true
+      register: instance
+
+    - set_fact:
+        volume_id: '{{ instance.instances[0].block_device_mappings[0].ebs.volume_id }}'
+        instance_id: '{{ instance.instances[0].instance_id }}'
+        device_name: '{{ instance.instances[0].block_device_mappings[0].device_name }}'
+
+#    JR: Check mode not supported
+#    - name: Take snapshot (check mode)
+#      ec2_snapshot:
+#        instance_id: '{{ instance_id }}'
+#        check_mode: true
+#        snapshot_tags:
+#          Test: '{{ resource_prefix }}'
+#      register: result
+#    - assert:
+#        that:
+#          - result is changed
+
+    - name: Take snapshot of volume
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+      register: result
+
+    # The Name tag is created automatically as the instance_name; ie the resource_prefix
+    - name: Get info about snapshots
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - result is changed
+          - info_result.snapshots| length == 1
+          - info_result.snapshots[0].snapshot_id == result.snapshot_id
```

Comment:
While it's not documented (ec2_snapshot doesn't document any return values) ec2_snapshot looks like it also returns volume_id, volume_size and tags.

Since there's currently **no** return documentation I'm guessing people (like me) would simply look at the output of -vvv and rely on consistent output.   As such I guess we should test the other output values to ensure that when someone has chance to migrate from boto2 to boto3 they don't get dropped.

Reply:
Now with bonus return docs! (It's maybe overloading this PR a bit, but it's a small enough addition.)